test/core/event_engine/thread_pool_test.cc

```diff
@@ -99,6 +99,19 @@ TEST(ThreadPoolDeathTest, CanDetectStucknessAtFork) {
       "Waiting for thread pool to idle before forking");
 }
 
+void ScheduleTwiceUntilZero(ThreadPool* p, int n) {
+  if (n == 0) return;
+  p->Add([p, n] {
+    ScheduleTwiceUntilZero(p, n - 1);
+    ScheduleTwiceUntilZero(p, n - 1);
+  });
+}
+
+TEST(ThreadPoolTest, CanStartLotsOfClosures) {
```

Comment:
For posterity, it's worth a comment explaining that the previous impl would crash trying to spawn 1 million threads if it were left unthrottled.

Reply:
done